### ISSUE TITLE
Side panel name wrong

### ISSUE THREAD

**Issue by octaeder** [texstudio-org/texstudio#2340]
Open the side panel, choose the symbols panel (or any of them that is not at the top of the secondary toolbar -- which is normally the structure panel). Close txs. The index of the active panel is saved in the ini file. Thus starting txs will restore that panel. But the side panel name shows always "Structure" at that point.

![grafik](https://user-images.githubusercontent.com/102688820/169394785-b7757904-7965-4f1a-a7ee-c27eee55dc60.png) Choose **symbols** panel, then exit txs

![grafik](https://user-images.githubusercontent.com/102688820/169394901-3decef3d-3521-46fe-b54b-7f8772cea336.png) txs remembers the **symbols** panel, but the side panel name shows **Structure**!

If any of the panels is hidden that is on top of the current selected one, then txs fails to restore the same panel and starts with a different one (s. also next comment):

![grafik](https://user-images.githubusercontent.com/102688820/169506027-cd72428a-4aac-4dbd-a2b7-5da42fa67ba2.png) Symbols is active panel, Strucure panel is hidden, then exit txs

txs saves GUI\sidePanel\currentPage=2 in the ini file. Counting starts with 0=TOC, 1=Bookmarks, 2=Symbols in side panel's vertical toolbar (not in the _widgets_ list in the context menu).

![grafik](https://user-images.githubusercontent.com/102688820/169506305-33d141b5-c7bb-4a0f-b7b5-e4625cea34a0.png) After restart of txs the Bookmarks panel is active instead of Symbols (and the side panel name is wrong)
Be aware that Bookmarks is second counting in the _widgets_ list.

### Environment
- TeXstudio: 4.2.3 and devs
- Qt: 6
- OS: Win10
- TeX distribution: miktex

**Comment by octaeder** [comment]
I observed that my solution did not work when the structure panel (it is the first one) is hidden. In this case the current txs (without my changes) already shows a slight error in the gui. If you uncheck (hide) the structure panel then the next (now first) panel (TOC) is activated:
![grafik](https://user-images.githubusercontent.com/102688820/169416406-bc86eaef-d799-4883-95f5-a28b404456f6.png)
Compare this when I (re)check the structure panel:
![grafik](https://user-images.githubusercontent.com/102688820/169416714-1359a396-4562-46a4-8d7b-666949e555f0.png)
In the first case the blue background of the icon (showing that this panel is the current showing) is missing.

Problems:
1) On changes to the side bar the code tries to activate the first panel but in the (full) list of all panels. In fact it has to be the first in the (shorter) list of visible (checked) panels.
2) The index stored in the ini file is the index in the shorter list, but on startup of txs the code (re)activates the panel with that index in the full list (and it should set up the side panel name here). In this the previous (one step up) panel is selected (should say "if present").

**Comment by octaeder** [comment]
texstudio.cpp l. 711ff (in function _setupDockWidgets_):
```
leftPanel->showWidgets();
// restore selected view in sidepanel
int viewNr=configManager.getOption("GUI/sidePanel/currentPage", 0).toInt();
leftPanel->setCurrentWidget(leftPanel->widget(viewNr));
```
_widget(viewNr)_ selects from the full _widgets_ list (s. toolwidgets.cpp, l. 582). After this the side panel name must be set to the current widget name with something like ```leftPanel->setWidgetText()```. Otherwise we always start with the side panel name "Structure" (s. _showWidgets_ below).

toolwidgets.cpp l. 553f (in function _showWidgets_, this function is called in the code above):
```
if (!widgets.empty()) //name after active (first) widget
    emit titleChanged(widgets.first()->property("Name").toString());
```
Here the first item out of the full _widgets_ list is taken which may be not visible (unchecked). Then the blue background is missing.

What is the short list? In toolwidgets.cpp, l. 437 a stackedWidget object stack is created. Function _showWidgets_ shows (l. 543f)
```
if (!hiddenWidgetsIds.contains(widgetId(widgets[i]))) {
			stack->addWidget(widgets[i]);
```
So I believe that the stack contains the panel widgets that are visible/offered in the side panel.

**Comment by octaeder** [comment]
@sunderme I assume that my analysis is helpfull. But I struggle with C++ and how to reference the stack. Can you give some hints? Please set label "help wanted".

**Comment by octaeder** [comment]
@sunderme Please remove label "help wanted". I'm making good progress: After restart of txs correct toolbar widget is selected.

![grafik](https://user-images.githubusercontent.com/102688820/169655696-f814f4bf-0d8e-4aab-b9cc-37a1f9f3df12.png) Some tools are hidden, current tool is Symbols, end txs

The ini file shows GUI\sidePanel\currentPage=1 (tool index in the toolbar).

![grafik](https://user-images.githubusercontent.com/102688820/169655858-0bfeee4a-8b66-4085-bf5a-5089307de6e2.png) After restart of txs tool Symbols is active (so correct)

**Comment by octaeder** [comment]
@sunderme Please find my test results here (direct comparison). I'll come up with a PR soon.

Side by side comparison ![grafik](https://user-images.githubusercontent.com/102688820/169662630-4d03c982-58b0-4a9a-a5f3-398891cd76dc.png) (left) vs. compiled ![grafik](https://user-images.githubusercontent.com/102688820/169662316-fdd77242-7f65-4783-8810-59f21350db02.png) (right):
1. In last txs session Structure was hidden, Symbols was active. Now start txs:
![grafik](https://user-images.githubusercontent.com/102688820/169661362-3b1f9316-86d8-484b-b0ac-994b630e6f4c.png) ________ ![grafik](https://user-images.githubusercontent.com/102688820/169661413-42b5923a-8bcd-4e0e-a054-06e87ccc6365.png)
2. Hide Tool Macros (be aware of the blue frame). By definition the first tool is active:
![grafik](https://user-images.githubusercontent.com/102688820/169661468-6ea7a3bf-e7cb-458c-bb09-e0fd67444c46.png)  ________ ![grafik](https://user-images.githubusercontent.com/102688820/169661496-5ad875dc-2a36-411b-87d8-bac11d640b62.png)
 3. Hide all Tools:
![grafik](https://user-images.githubusercontent.com/102688820/169661579-bd4b3f13-23e6-4540-a12a-ae3daedf3eaa.png)  ________ ![grafik](https://user-images.githubusercontent.com/102688820/169661595-e664e09f-07df-4200-804c-9f61da88e803.png)
4. End and restart txs without tools:
![grafik](https://user-images.githubusercontent.com/102688820/169661778-08491f33-163e-4d49-9547-8b6509067736.png)  ________ ![grafik](https://user-images.githubusercontent.com/102688820/169661700-8959424a-dec1-45e6-8fb7-240287e58061.png)
After the crash you can't restart txs anymore without fixing the ini file such that one ore more tools will be present. S. ini file: Symbols\hiddenlists=xymatrix|beamer|asymptote|tikz|metapost|pstricks|brackets|structureTreeWidget|symbols
5. Hide active tool Symbols (just to check if something new happens, but works as before):
![grafik](https://user-images.githubusercontent.com/102688820/169662186-2b8a7e46-d31f-4513-a5b2-1d3cae2fbd9f.png)  ________ ![grafik](https://user-images.githubusercontent.com/102688820/169662123-16e17824-b12a-44fe-837f-5c534074cb2e.png)

Also tested with version ![grafik](https://user-images.githubusercontent.com/102688820/169662853-bc0359d8-2afb-424a-97f3-8826153be36a.png) from PR #2330. Results where the same as with 4.2.3 (on the left).

**Comment by sunderme** [comment]
to be honest, I am not thrilled that PR description and PR are separate. This makes analysing the stuff, especially later on unnecessarily cumbersome.
Is the crash still occuring with the PR ?

**Comment by octaeder** [comment]
Can you show some PR or else that shows me how to do best with so many documentation I give? Issue and PR link to each other, so both informations can easily be found. Maybe you can tell me a bit of your way of working, so I can see what makes it cumbersome and what I can do better.

Not sure, why you ask for the crash: Image to the left shows a crash, while image to the right shows txs with an empty left panel as expected. Besides this, starting the new version, adding some tools and exiting txs allowed me to restart old version without fiddling in the ini file. I noted the key/value from the ini file just in case someone really needs it. So, for me things get better.

**Comment by sunderme** [comment]
okay, I misinterpreted the images.
as a PR example, maybe #1189

**Comment by octaeder** [comment]
resolved by [254cfc6](https://github.com/texstudio-org/texstudio/commit/254cfc69201d6462dc6b7a71d60b6c31ef8ccf6b)
